.github/workflows/main.yml

```
@@ -0,0 +1,62 @@
+# This workflow will install Python dependencies, run tests and lint with a
+# variety of Python versions. For more information see:
+# https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Pytest with Flake8
+
+on: [push, pull_request]
+
+jobs:
+  build:
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]  # TODO(76): add windows-latest
+        python-version: ["3.9", "3.10"]
+        numpy_ver: ["latest"]
+        include:
+          - python-version: "3.8"
+            numpy_ver: "1.20.3"
+            os: ubuntu-latest
+          - python-version: "3.6.8"
+            numpy_ver: "1.19.5"
+            os: "ubuntu-20.04"
+
+    name: Python ${{ matrix.python-version }} on ${{ matrix.os }} with numpy ${{ matrix.numpy_ver }}
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install standard dependencies
+      run: |
+        pip install -r test_requirements.txt
+        pip install -r requirements.txt;
+
+    - name: Install NEP29 dependencies
+      if: ${{ matrix.numpy_ver != 'latest'}}
+      run: |
+        pip install numpy==${{ matrix.numpy_ver }}
+
+    - name: Set up pysat
+      run: |
+        mkdir pysatData
+        python -c "import pysat; pysat.params['data_dirs'] = 'pysatData'"
+
+    - name: Test PEP8 compliance
+      run: flake8 . --count --select=E,F,W --show-source --statistics
```

```suggestion
      run: flake8 . --count --select=D,E,F,H,W --show-source --statistics
```
This is the standard elsewhere, though we may get a lot of style feedback